app/Models/Multiplayer/UserScoreAggregate.php

```diff
@@ -128,11 +134,47 @@ public function removeRunningTotals()
         }
     }
 
+    public function room()
+    {
+        return $this->belongsTo(Room::class);
+    }
+
+    public function scopeForRanking($query)
+    {
+        return $query
+            ->where('completed', '>', 0)
+            ->whereHas('user', function ($userQuery) {
+                $userQuery->default();
+            })
+            ->orderBy('total_score', 'DESC')
+            ->orderBy('last_score_id', 'ASC');
+    }
+
     public function updateUserAttempts()
     {
         $this->increment('attempts');
     }
 
+    public function user()
+    {
+        return $this->belongsTo(User::class, 'user_id');
+    }
+
+    public function userRank()
+    {
+        if ($this->total_score === null || $this->last_score_id === null) {
+            return;
+        }
+
+        $query = $this->forRanking()
```

Comment:
This needs filter for `room_id`. Also it's weird seeing `$this->` when it should be `static::`